### PR TITLE
Add support for the L20n's FTL format

### PR DESCRIPTION
**Pull request by mathjazz**
@zbraniecki r?

I'm leaving the original L20n file format support in the codebase for now. We should remove it eventually.

### CODE REVIEW

File: pontoon/sync/formats/ftl.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                )
+
+        with codecs.open(path, 'r', 'utf-8') as resource:
+            self.structure = FTLParser().parseResource(resource.read())
+
+        def get_comment(obj):
+            return [obj['comment']['content']] if obj['comment'] else []
+
+        def parse_entity(obj, section_comment=[]):
+            if obj['value'] and obj['value']['type'] == 'Pattern':
```
Comment (zbraniecki) [inline pontoon/sync/formats/ftl.py]: Why not `if obj['type'] == 'Entity'` ?

Comment (zbraniecki) [inline pontoon/sync/formats/ftl.py]: Ah, I see. You're just checking if the entity has a value. A value is always a Pattern

File: pontoon/sync/formats/ftl.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        with codecs.open(self.source_resource.path, 'r', 'utf-8') as resource:
+            structure = FTLParser().parseResource(resource.read())
+
+        def serialize_entity(obj, entities):
+            entity_id = obj['id']['name']
+            translations = self.entities[entity_id].strings
+
+            if obj['value'] and obj['value']['type'] == 'Pattern':
+                if translations:
+                    obj['value']['elements'][0]['value'] = translations[None]
```
Comment (zbraniecki) [inline pontoon/sync/formats/ftl.py]: I don't think I understand that. If you're recreating a Pattern, you need to also recreate it's source (`obj['value']['source']`), and you probably should regenerate the whole `elements` list.

**Comment by zbraniecki** [comment]
r+
You seem to have some failing tests for l20n format?

File: pontoon/sync/formats/__init__.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
     '.ini': silme.parse_ini,
     '.inc': silme.parse_inc,
     '.l20n': l20n.parse,
+    '.ftl': ftl.parse,
```
Comment (jotes) [inline pontoon/sync/formats/__init__.py]: I guess it would be good idea to add a small comment about the future of support l20n format.

**Comment by mathjazz** [comment]
@zbraniecki This is now ready for a second look.

File: pontoon/sync/formats/__init__.py (hunk 3 above)
Comment (mathjazz) [inline pontoon/sync/formats/__init__.py]: Yeah. Or maybe we can just remove the old l20n format support? @zbraniecki?

Comment (zbraniecki) [inline pontoon/sync/formats/__init__.py]: you can definitely remove the old format.

File: pontoon/sync/formats/ftl.py [hunk 4]
```diff
@@ -103,7 +103,8 @@ def serialize_entity(obj, entities):
             translations = self.entities[entity_id].strings
 
             if translations:
-                obj['value']['elements'][0]['value'] = translations[None]
+                source = translations[None]
+                obj['value'] = ParseContext(source).getPattern().toJSON()
```
Comment (zbraniecki) [inline pontoon/sync/formats/ftl.py]: wow! that's a neat approach, I like it :)

**Comment by zbraniecki** [comment]
r+

**Pull request merged**